src/engine/strat_engine/backstore/metadata.rs

```diff
@@ -505,7 +561,7 @@ mod mda {
                 )))?;
                 f.write_all(&hdr_buf)?;
                 f.write_all(data)?;
-                f.flush()?;
+                f.sync_all()?;
```

Comment:
Looks like this change and also change immediately above do not belong in this PR. AFAICT, they should have gone in separately, with introduction of SyncAll trait. Please move these and any other changes in the same situation into a separate PR.